index.js

```diff
@@ -59,48 +48,62 @@ fs.readFile(file, 'utf8', (err, data) => {
            - these are in clipping[1], need to extract the relevant parts
         */
 
-        // when called this function creates a new book Object using the details above
-       function createBook (title, highlight) {
-        let book = {
-            title: title,
-            highlights: [highlight]
-        }
-        return book;
-       }
-
-        if (!books.length) {
-            //if books is empty, call createBooks and push the created Object to books
-            let book = createBook(title, highlight);
-            books.push(book);
+        if (books[title]) {
+          books[title].push(highlight);
         }
         else {
-            //if array is not empty then we must search if a book exists and find its index
-            searchBooks(books, title);
-
-            if (bookIndex >= 0) {
-                //if the book exists then we only need to push the highlight to the highlights array
-                books[bookIndex].highlights.push(highlight);
-            }
-            else {
-                //if book does not exist then create it
-                let book = createBook(title, highlight);
-                books.push(book);
-            }
-        }   
+          books[title] = [highlight];
+        }
     }
 
     // Loop through each book and write its contents to txt file
 
-    books.forEach((book) => {
-        txtFilePath = pathName + book.title + '.txt';
-        for (let i = 0; i < book.highlights.length; i++) {
-            fs.appendFile(txtFilePath, book.highlights[i] + "\n\n", (error) => {
```

Comment:
I can foresee users of this script (including myself from testing it) running it multiple times. Constantly appending tot he same files will result in duplicate entries. Actually the thing I wrote on the right doesn't resolve the issue properly. But something that writes (thus overwriting) or appends conditionally would be great here.